### PR TITLE
Excavator:  Switch to JUnit 5 to parallelize tests and speed up CI

### PR DESCRIPTION
**Pull request by schlosna**
## Before this PR
JUnit 5 Excavator https://github.com/palantir/syntactic-paths/pull/11 is failing

## After this PR

==COMMIT_MSG==
Excavator:  Switch to JUnit 5 to parallelize tests and speed up CI
==COMMIT_MSG==

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -49,5 +49,19 @@ allprojects {
...
+        }
+        maxParallelForks Runtime.getRuntime().availableProcessors()
+        testLogging {
+            events "passed", "skipped", "failed"
+            exceptionFormat "full"
+            showCauses true
+            showExceptions true
+            showStackTraces true
+        }
+    }
```
Comment (carterkozak) [inline build.gradle]: huh, I'd have expected this to be handled out of the box by baseline

File: versions.props [hunk 2]
```diff
@@ -1,7 +1,7 @@
 com.fasterxml.jackson.core:jackson-databind = 2.13.3
 com.google.guava:guava = 31.1-jre
 com.palantir.safe-logging:* = 3.0.0
-junit:junit = 4.12
 org.assertj:assertj-core = 3.23.1
+org.junit.jupiter:* = 5.5.0
```
Comment (carterkozak) [inline versions.props]: Might as well jump to latest 5.9.0, 5.5.0 was released a bit over three years ago!

Comment (schlosna) [reply versions.props]: Good call, I had copied this from an older project template. Will update when I can `write-locks` (note to self, look at nit bot for OSS repos)

File: build.gradle (hunk 1 above)
Comment (schlosna) [reply build.gradle]: I can rip this out, I had copied over from tritium to see all the tests running.

**Review by carterkozak — APPROVED** [review]

**Pull request merged**